lib/ui/widgets/dialogs/network_dialog.dart

```diff
@@ -76,145 +64,110 @@ class NetworkDialog {
-                                              localizations.enterEndpointBlank;
-                                          FocusScope.of(context).requestFocus(
-                                            endpointFocusNode,
-                                          );
-                                        });
-                                      } else {
-                                        if (Uri.parse(
-                                              endpointController.text,
-                                            ).isAbsolute ==
-                                            false) {
-                                          setState(() {
-                                            endpointError = localizations
-                                                .enterEndpointNotValid;
-                                            FocusScope.of(context).requestFocus(
-                                              endpointFocusNode,
-                                            );
-                                          });
-                                        } else {
-                                          await ref
-                                              .read(
-                                                SettingsProviders
-                                                    .settings.notifier,
-                                              )
-                                              .setNetwork(
-                                                NetworksSetting(
-                                                  network:
-                                                      selectedNetworkSettings
-                                                          .network,
-                                                  networkDevEndpoint:
-                                                      endpointController.text,
-                                                ),
-                                              );
-
-                                          Navigator.pop(context);
-                                        }
-                                      }
-                                    },
+                        return _NetworkDialogCustomInput(
+                          endpointFocusNode: endpointFocusNode,
+                          endpointController: endpointController,
+                          endpointError: endpointError,
+                          onSubmitNetwork: () async {
+                            void setError(String errorText) {
+                              setState(() {
+                                endpointError = errorText;
+                                FocusScope.of(context).requestFocus(
+                                  endpointFocusNode,
+                                );
+                              });
+                            }
+
+                            Future<List<Node>> getNodeFromNetwork(
```

Comment:
These business functions could be Family Async Providers declared in a new `/lib/application/network` directory.


Example using `Riverpod` generator.

<details>
<summary>
/lib/application/network/provider.dart
</summary>

```dart
@riverpod
List<Node> _networkNodes(
  _NetworkNodesRef ref, {
  required AvailableNetworks network,
}) {
    final networkSetting = NetworksSetting(
      network: network,
      networkDevEndpoint: '',
    );

    final link = networkSetting.getLink();

    final nodeListMain = await ApiService(
      link,
    ).getNodeList();

    return nodeListMain;
}

@riverpod
bool _isReservedNodeUri(
  _IsReservedNodeUri ref, {
  required Uri uri,
}) {
  final nodeListMain = await ref.watch(_networkNodesProvider(network AvailableNetworks.archethicMainNet));
  final nodeListTest = await ref.watch(network: _networkNodesProvider(AvailableNetworks.archethicTestNet));

  return nodeListMain.followedBy(nodeListTest).any(
      (node) =>
          node.ip == uri.host &&
          node.port == uri.port,
    );
}

abstract class NetworkProvider {
  static final isReservedNodeUri = _isReservedNodeUriProvider;
}
```
</details>


<details>
<summary>
NetworkDialog
</summary>

```dart
....
   onSubmitNetwork: () async {
      if (await ref.readNetworkProvider.isReservedNodeUri(uriInput)) {
          setError(
            localizations.enterEndpointUseByNetwork,
          );
          return;
       }

        await ref
            .read(
              SettingsProviders.settings.notifier,
            )
            .setNetwork(
              NetworksSetting(
                network: AvailableNetworks.archethicDevNet,
                networkDevEndpoint: textCleaned,
              ),
            );
   }
...
```
</details>